src/models/Dialog.js

```diff
@@ -38,4 +38,11 @@ const DialogSchema = new Schema(
   }
 );
 
+DialogSchema.virtual("approval_rate").get(function CalculatesApprovalRate() {
```

Comment:
Would it be possible to extract this function to the **utils** folder and just use it here? This would greatly facilitate the addition of unit tests.

research on this please